src/util/middleware.ts

```diff
@@ -8,32 +8,35 @@
  * of the MIT license. See the LICENSE file for details.
  */
 
-import Config from '../config';
-import { RequestObj } from '../config/interfaces';
+import { RequestMiddleware, RequestObj } from '../config/interfaces';
 import { ActionTypes } from '../config/enums';
 
 /**
  * @function middlewareWrapper - A "Node" and "Redux" style middleware that is called just before
  * the request is made from the SDK. This allows you access to the request for modification.
- * @param request - A request object container the URL and the Request Init object
- * @param type - A "Redux" style type that contains the serialized action
- * @param payload - The payload of the action that can contain metadata
+ * @param middleware - The middleware to run the request through
+ * @param request - A request object container of the URL and the Request Init object
+ * @param action - The action object that is passed into the middleware communicating intent
+ * @param action.type - A "Redux" style type that contains the serialized action
+ * @param action.payload - The payload of the action that can contain metadata
  */
-// eslint-disable-next-line
-function middlewareWrapper(request: RequestObj, type: ActionTypes, payload?: any): RequestObj {
-  const { middleware } = Config.get();
-
+function middlewareWrapper(
+  middleware: RequestMiddleware[] | undefined,
```

Comment:
The only thing i'm having a hard time wrapping my head around is why we want the middleware to be the first argument to the middlewareWrapper.

This feels ergonomically weird to me as it can be `undefined`. 

What would the benefits of this signature be to say a signature where we curried a function expecting middleware?

Say:

`middlewareWrapper(request: RequestObj, type: ActionTypes, payload?: any) => (Middleware) => RequestObj;`

We may not need to curry it at all, but the ability to pass undefined as a first argument feels a little weird. But if we have good reason for it i'm ok.